src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,97 @@
 public class StringManipulation implements StringManipulationInterface {
+    private String string;
 
     @Override
     public String getString() {
-        return null;
+        return string;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        String[] preFilteredResult = string.split(" ");
+        int count = 0;
+        
+        for (String result : preFilteredResult) {
+            if (!result.isEmpty()) {
+                count++;
+            }
+        }
+        
+        return count;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        if (n <= 0) {
+            throw new IllegalArgumentException();
+        } else if (n > string.length()) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        for (int i = 0; i < string.length(); i++) {
+            if ((i + 1) % n == 0) {
+                if (maintainSpacing) {
+                    stringBuilder.append(" ");
```

Comment:
I think changing .append to .deleteCharAt() could add some optimization by possibly avoiding creating a new string for each character removal